client/client.go

```diff
@@ -373,26 +379,23 @@ func (c *client) tsLoop() {
 							select {
 							case tsDeadlineCh.(chan deadline) <- dl:
 							case <-loopCtx.Done():
-								cancel()
 								return
 							}
 							opts = extractSpanReference(requests[:pendingPlus1], opts[:0])
 							err = c.processTSORequests(stream, dc, requests[:pendingPlus1], opts)
 							close(done)
 						case <-loopCtx.Done():
-							cancel()
 							return
 						}
+						// If error happens during tso stream handling, reset stream and run the next trial.
 						if err != nil {
 							select {
 							case <-loopCtx.Done():
-								cancel()
 								return
 							default:
 							}
 							log.Error("[pd] getTS error", zap.String("dc-location", dc), errs.ZapError(errs.ErrClientGetTSO, err))
 							c.ScheduleCheckLeader()
-							cancel()
```

Comment:
I think this 'cancel()' should be retained.

Reply:
updated.